packages/compass-workspaces/src/services/workspaces-storage.tsx

```diff
@@ -0,0 +1,102 @@
+import { createServiceLocator } from '@mongodb-js/compass-app-registry';
+import {
+  IUserData,
+  type ReadAllResult,
+  z,
+} from '@mongodb-js/compass-user-data';
+import React, { useContext, useRef } from 'react';
+
+/**
+ * Schema for saving workspace tab state to user data
+ */
+
+// Define schema for collection subtab
+const CollectionSubtabSchema = z.enum([
+  'Documents',
+  'Aggregations',
+  'Schema',
+  'Indexes',
+  'Validation',
+  'GlobalWrites',
+]);
+
+// Define schema for workspace tab type
+const WorkspaceTabTypeSchema = z.enum([
+  'Welcome',
+  'My Queries',
+  'Data Modeling',
+  'Shell',
+  'Databases',
+  'Performance',
+  'Collections',
+  'Collection',
+]);
+
+// Define schema for a workspace tab
+export const WorkspaceTabSchema = z.object({
+  id: z.string(),
+  type: WorkspaceTabTypeSchema,
+  connectionId: z.string().optional(),
+  namespace: z.string().optional(),
+  initialQuery: z.record(z.any()).optional(),
+  initialAggregation: z.record(z.any()).optional(),
+  initialPipeline: z.array(z.record(z.any())).optional(),
+  initialPipelineText: z.string().optional(),
+  editViewName: z.string().optional(),
+  initialEvaluate: z.union([z.string(), z.array(z.string())]).optional(),
+  initialInput: z.string().optional(),
+  subTab: CollectionSubtabSchema.optional(),
+});
```

Comment:
Sort of relevant to my comment above: the schema here should be a tuple of values by workspace type, similar to how it's defined in the types file. That way zod will make sure that we will never end up with workspaces where properties are not matching the type